controllers/auth.js

```diff
@@ -0,0 +1,50 @@
+const User = require("../models/User.js");
+const Profile = require("../models/Profile");
+const { StatusCodes } = require("http-status-codes");
+const { BadRequestError, UnauthenticatedError } = require("../errors");
+
+const register = async (req, res) => {
+  const checkUser = await User.findOne({ email: req.body.email });
+
+  if (checkUser) {
+    res.status(StatusCodes.BAD_REQUEST).json({ error: "Duplicate", msg: "User already exist!" });
+    return;
+  }
+
+  const user = await User.create(req.body);
+ 
+  await Profile.create({
+    userName: req.body.userName,
+    email: req.body.email,
+    city: "",
+    country: "",
+  });
+
+  const token = user.createJWT();
+
+  res.status(StatusCodes.CREATED).json({ user: user, token });

```

Comment:
Don't return the whole user object in the response because that's going to expose the password.

```suggestion
  res.status(StatusCodes.CREATED).json({ user: { email: user.email, userName: user.userName }, token });
```